### PR TITLE
419 fix missing exception

### LINKED ISSUE #419: Instance during CI still in preflight after timeout
shakenfist_ci.base.TimeoutException: Instance cc2ddb97-c543-428c-a49b-a64060423342 was not created in a reasonable time ({'block_devices': None, 'console_port': 32707, 'cpus': 1, 'disk_spec': [{'base': 'https://nosuch.shakenfist.com/foo', 'size': 8, 'type': 'disk'}], 'error_message': None, 'memory': 1024, 'name': 'cirros', 'namespace': 'ci-cacheimage-tidrwrzb', 'node': 'sfci-mcamyc.c.mikal-269605.internal', 'node_history': [], 'placement_attempts': 0, 'power_state': 'initial', 'requested_placement': None, 'ssh_key': None, 'state': 'preflight', 'state_updated': 1602991219.5415065, 'user_data': None, 'uuid': 'cc2ddb97-c543-428c-a49b-a64060423342', 'vdi_port': 35261, 'video': {'memory': 16384, 'model': 'cirrus'}})

### PR DESCRIPTION
**Pull request by mandoonandy**
Find missing image fetch exception and add event
Fix image task bugs and add image events endpoint

Requires https://github.com/shakenfist/client-python/pull/9

Resolves #419

### CODE REVIEW

File: deploy/shakenfist_ci/base.py [hunk 1]
```diff
@@ -132,6 +140,33 @@ def _await_event(self, instance_uuid, operation, message=None, after=None):
...
                 after, instance_uuid, operation, message))
 
+    def _await_image_download_success(self, url):
+        return self._await_image_event(
+            url, 'fetch', 'success')
+
+    def _await_image_download_error(self, url):
+        return self._await_image_event(
+            url, 'fetch', 'Name or service not known')
+
```
Comment (mikalstill) [inline deploy/shakenfist_ci/base.py:150]: The only catch with this approach is it means failing tests always take 300 seconds to timeout, whereas if we asserted that waiting for a success means that there are no failures then we could fail faster?

**Review by mikalstill — COMMENTED** [review]
Small comment.

File: deploy/shakenfist_ci/base.py (hunk 1 above)
Comment (mandoonandy) [reply deploy/shakenfist_ci/base.py:150]: Yes, there is a better way. I sort that out after dinner.

**Comment by mikalstill** [comment]
CI here should now pass because the client change has been landed. We're not mature enough to be able to express CI dependancies yet.

**Comment by mikalstill** [comment]
CI is still unhappy... https://jenkins.shakenfist.com/job/shakenfist/view/change-requests/job/PR-423/2/execution/node/19/log/

**Comment by mandoonandy** [comment]
A few changes:

Fixed the tests to error immediately upon the wrong event.

Fixed queue handling: during image fetch exception handling, I think there was confusion of urllib HTTPError with shakenfist.exception.HTTPError

**Major change:** virt.py create() now only creates the virtual instance, it does not fetch any disk images. This removes the potential download errors from that code section. Instances are only created after a fetch so this step was redundant. (Let me know if I made a gross error?!)

Future change for Images - upon instantiation: check files exist or flag that fetch is required.

**Comment by mikalstill** [comment]
Nice. I thought I'd already removed the fetch from virt.Instance.create(), so I am fine with that.

**Review by mikalstill — APPROVED** [review]
LGTM, thanks!

**Pull request merged**